lib/classes/Variables.test.js

```diff
@@ -1359,6 +1359,49 @@ module.exports = {
         .populateProperty(property)
         .should.eventually.eql('my stage is dev');
     });
+
+    it('should support ${file(...)} syntax', () => {
+      // eslint-disable-next-line no-template-curly-in-string
+      const property = '${file(~/somedir/../config.yml)}';
+
+      const expectedFileName = `${os.homedir()}/somedir/../config.yml`;
+      const configYml = {
+        foo: 'bar',
+      };
+      const fileExistsStub = sinon.stub(serverless.utils, 'fileExistsSync').returns(true);
+      const realpathSync = sinon.stub(fse, 'realpathSync').returns(expectedFileName);
+      const readFileSyncStub = sinon.stub(serverless.utils, 'readFileSync').returns(configYml);
+
+      return serverless.variables
+        .populateProperty(property)
+        .then(valueToPopulate => {
+          expect(realpathSync).to.not.have.been.called;
+          expect(fileExistsStub).to.have.been.calledWithMatch(expectedFileName);
+          expect(readFileSyncStub).to.have.been.calledWithMatch(expectedFileName);
+          expect(valueToPopulate).to.deep.equal(configYml);
+        })
+        .finally(() => {
+          realpathSync.restore();
+          readFileSyncStub.restore();
+          fileExistsStub.restore();
+        });
+    });
```

Comment:
I added a higher level test here because there was no test covering that `${file(...)}` is actually parsed correctly. When I changed the regex, initially the tests passed but it was a false positive: that syntax was broken. So I added that test and iterated on the regex.